### PR TITLE
First approach to markdownify title

### PR DESCRIPTION
**Pull request by Syralist**
This is my approach to use markdownified titles. Implements #179 
Not sure if this is the best way to do it, or if there is a more central place to apply markdownify just once?

### CODE REVIEW

**Comment by palant** [comment]
I'd probably extend `partials/utils/title.html` to make it return `htmlTitle` as well, the logic liekly won't stay this simple. In particular, I'd assume that @reuixiy wants to run the titles through `partials/utils/markdownify.html` rather than plain `markdownify`, e.g. for sake of emojis. And this functionality should probably be behind a config parameter - and be it so that existing users can leave it off and don't worry about breakage.

The question is whether category/tag titles should receive the same treatment - I'd say that they should, for consistency reasons. This requires changes in a bunch more places.

And you are probably aware of it but just in case: page title is used in some more templates, e.g. `partials/components/post-share.html`, `index.sectionrss.xml` and `index.sectionatom.xml` (these need so start using `$rawTitle`).

**Comment by reuixiy** [comment]
I agree with @palant, it would be better if we extend the existing `partials/utils/title.html`.

**Comment by Syralist** [comment]
I suggest adding htmlUnescape to title and rawTitle.
Otherwise there will be things like &ndash; in the title. In the browser title and the RSS this would be tolerable but unaesthetic. But in the share-links it breaks the links causing the titles to be abbreviated.

File: layouts/partials/utils/title.html [hunk 1]
```diff
@@ -1,27 +1,30 @@
+{{- $ := index . "$" -}}
+{{- $rawTitle := .title -}}
```
Comment (palant) [inline layouts/partials/utils/title.html:2]: It's rather non-obvious now what this template returns, with the final calculations being performed in the return statement. How about making things less confusing and naming this variable `$htmlTitle`? $rawTitle can then be defined afterwards as `|plainify|htmlUnescape` version of it. And `$title` would be based on `$rawTitle` again. Or was allowing HTML tags in `.Site.Title` intentional?

**Comment by reuixiy** [comment]
@palant My apologies, I merged this in a hurry. Could you create another pull request to improve this?

**Pull request merged**